tests/migration/online_migration/yast2_migration.pm

```diff
@@ -220,10 +220,13 @@ sub run {
             send_key 'alt-t';
         }
     }
-    assert_screen ['yast2-migration-installupdate', 'yast2-migration-proposal'], 700;
-    if (match_has_tag 'yast2-migration-installupdate') {
+    if (check_screen('yast2-migration-installupdate', 700)) {    # Not all cases have install update message
```

Comment:
Please find a better solution than wasting 700s (!). I don't see why your situation would deny the use of multi-tag.

Reply:
I'm testing more cases to reduce the time here. 
For the multi-tag, the license agreement I added here is appeared between yast2-migration-installupdate and yast2-migration-proposal/summary, and the license must be shown, it is not optional. The flow here is:
firstly, installupdate (maybe have it, some cases don't have it);
then, license agreement (all the cases must print it, otherwise we should file a bug);
and then migration proposal (must have it, but it is in the same page with migration_summary, two tags in the same page)

Reply:
Sorry, I don't see why you should not be able to use a better approach. This brought me to the idea to introduce an automatic code style check: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12821

Reply:
If using multi-tag, when it matches one of the three tags, the case will pass, right? In this situation, there are three tags, it should match tag1, tag2 and tag3; or it should match tag2 and tag3, so multi-tag is not suitable here.

Reply:
> so multi-tag is not suitable here.

that's not correct. You can always find a way to express final states that you want to reach. A sane human operator would also not "wait for 1 or 2 or until 12 minutes have passed"

Reply:
In https://openqa.nue.suse.com/tests/6267218#step/yast2_migration/31 I can see that the test fails expecting "yast2-migration-installupdate" or "yast2-migration-proposal" but instead a license dialog shows up which can be looked for with "yast2_migration-license-agreement". Of course after handling the license agreement you are not done with the migration so I assume after accepting the license you still want to look for "yast2-migration-installupdate" or "yast2-migration-proposal", right? So you could use something like

```perl
assert_screen [qw(yast2-migration-installupdate yast2-migration-proposal yast2_migration-license-agreement)];
if (match_has_tag 'yast2_migration-license-agreement') {
    send_key 'alt-a';
    assert_screen [qw(yast2-migration-installupdate yast2-migration-proposal)];
}
…
```

https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/x11/kontact.pm#L21 provides a good elaborate example of how one can manage different dialogs that optionally pop up and in not defined order.

Reply:
Thanks for your advise. I'm going to have a try and update here.

Reply:
The case doesn't fit to use qw(), let me explain it here: installupdate is shown BEFORE license, while not all cases have installupdate message, so we need to use "if" on installupdate; License is must be shown for all cases, so we shouldn't use "if" for it. BTW, in https://openqa.nue.suse.com/tests/6267218#step/yast2_migration/31, we can see the license is shown, though we accept it, the next page is not installupdate, it would be proposal(same page with summary); For proposal, it is the same page with summary.

Reply:
Sorry, I don't understand your explanation why the quoted word list is not applicable. What has `qw()` have to do with the test flow? Have you looked at the kontact example with the loop? I think that could help

Reply:
Yes, you are right, finally I found a solution, thanks for your help.